receiver/faroreceiver/handler.go

```diff
@@ -0,0 +1,158 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package faroreceiver
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-logfmt/logfmt"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type Handler struct {
+	warnOnce sync.Once
+
+	logConsumer   consumer.Logs
+	traceConsumer consumer.Traces
+
+	logLogger   *zap.Logger
+	traceLogger *zap.Logger
+}
+
+var _ http.Handler = (*Handler)(nil)
+
+func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+
+	h.warnOnce.Do(func() {
+		if h.logConsumer == nil {
+			h.traceLogger.Warn("faroreceiver logs will not be consumed!")
+		}
+		if h.traceConsumer == nil {
+			h.logLogger.Warn("faroreceiver traces will not be consumed!")
+		}
+	})
+
+	var p Payload
+	if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if err := h.consumePayload(ctx, p); err != nil {
+		http.Error(w, "collector error", http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusAccepted)
+}
+
+func (h *Handler) consumePayload(ctx context.Context, p Payload) error {
+	logs, err := translate(ctx, p)
+	if err != nil {
+		return err
```

Comment:
Here, you could wrap (or just use a new error, if you are concerned about leaking details to the client), stating that:

```
return errors.New("failed to translate the payload into logs")
```